app/dao/notifications_dao.py

```diff
@@ -600,18 +599,16 @@ def dao_get_notifications_by_recipient_or_reference(service_id, search_term, not
         except InvalidEmailError:
             normalised = search_term.lower()
 
-    elif notification_type == LETTER_TYPE:
-        raise InvalidRequest("Only email and SMS can use search by recipient", 400)
-
     else:
```

Comment:
Two questions around this.

Why did we have an `else` statement originally here? If it's not sms, email or letter, surely something is up? Either that or we are allowing you to search across all notification types?

Secondly, I think I'd like it to see laid out as

```
if sms:
    do thing

if email:
    do other thing

if letter:
    do different thing

else:
    do this, with a comment explaining why the generic behaviour is appropriate for all notification types
```

It would be much more explicit this way about the behaviour for all notifications and letters. Because here it appears not are only are we adding functionality for searching for letters, but we are also changing the value of `normalised` for when we aren't searching by notification type

Reply:
For users who don’t have permission to see the dashboard all the notifications for their service appear on one page. Which means the search works across multiple notification types. There is an existing test for this: https://github.com/alphagov/notifications-api/blob/d88b20beeca799f920eb84fb1695f336e4fb5a29/tests/app/dao/notification_dao/test_notification_dao.py#L1314-L1318

When the only two searchable types of notification were `email` and `sms` then the lowest common denominator normalisation was to just lowercase the search term. But now that letters are searchable we need a normalisation that will also find letters. And that means smooshing spaces together, because our normalised addresses will have all their spaces smooshed together (in other words a search for ‘David M’ needs to match `davidm` in the database).

Is this worth adding as a comment in the code?

Reply:
I’ve added a comment, and removed the catch-all `else` in 64674be